nova/src/utils.rs

```diff
@@ -35,6 +35,13 @@ pub fn iter_bits_le(bytes: &[u8]) -> impl Iterator<Item = bool> + '_ {
         .flat_map(|byte| (0..8).map(move |bit| ((1 << bit) & byte) != 0))
 }
 
+#[macro_export]
+macro_rules! safe_log {
+    ($x:expr) => {
+        ($x - 1).checked_ilog2().unwrap_or(0) + 1
+    };
+}
+
```

Comment:
1. It's good to add a doc-comment explaining what it does: computes the number of bits needed to represent numbers in range
[0; x)
For example, given an array of N elements you need this many bits to represent each index.
2. It's not really safe:
```rust
let i = std::hint::black_box(0u64);
println!("{}", safe_log!(i));
```
prints 64 in release. I don't think we ever use it with zeros (it's asserted in most places), but it's either the name or the underflow check.

Reply:
Done.